internal/nginx-meshctl/commands/root.go

```diff
@@ -254,3 +258,28 @@ func getComponentVersions(k8sClient k8s.Client) (string, error) {
 
 	return str, nil
 }
+
+func newVersionClient(config *rest.Config, namespace string, timeout time.Duration) (*http.Client, string, error) {
+	gv := schema.GroupVersion{Group: "", Version: "v1"}
+	config.GroupVersion = &gv
+	config.APIPath = fmt.Sprintf("/api/v1/namespaces/%s/services/nginx-mesh-api:%d/proxy/version", namespace, mesh.ControllerVersionPort)
```

Comment:
Is there any way to hit this directly w/o having to proxy? Alternatively, could we get the version number from the configmap or something rather than make a call to the mesh API?

Reply:
The reason we call the API is because it returns the binary-embedded, immutable version. We can't rely on the ConfigMap version because it could be altered by someone.

Without ingress or a nodeport, I'm not sure how to hit this endpoint.

Reply:
That makes sense.

Reply:
What if we exposed the version as a Prometheus metric? That would be immutable and avoid adding an extra endpoint.

Reply:
Neat idea, but I don't think that would work. We return a full json block of the versions of sidecars, controller, and metrics pod.

Reply:
Ahh yeah, that won't work with that much information. Unless we could condense it down to one version per component.

We would have just been able to hit the /metrics endpoint and leave Prometheus out of it.

Reply:
I don't think we have a /metrics endpoint right now.

Reply:
Or maybe we do because of the controller-runtime library...I can't remember.